From 3279a18619ce8c3af0fd4a088fa950c1c2b74ac9 Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Tue, 06 May 2025 15:34:08 +0300
Subject: [PATCH] [SW-228057] lfs: add serdes lfs part csum to fw info

Copy Serdes LFS partition sha384 checksum from boot
descriptor to fw_info, so that arcmgmt will be able
to check it's validity

Change-Id: I62413599fcc55f0ea2c54d25565ee39343cd65b6
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/arch/gaudi3/common/validate_fw.c b/arch/gaudi3/common/validate_fw.c
index c7a4a45..52676ec 100644
--- a/arch/gaudi3/common/validate_fw.c
+++ b/arch/gaudi3/common/validate_fw.c
@@ -172,6 +172,7 @@
 	case HL_FW_IMG_SERDES_PARTITION_LOC:
 		fw_info.serdes_lfs_partition_offset = (uint32_t)(desc->loadaddr);
 		fw_info.serdes_lfs_partition_size = desc->img_size;
+		memcpy(fw_info.serdes_lfs_partition_checksum, desc->chksum, SHA384_SUM_LEN);
 		LOG_INF("Serdes partition: Offset = %08X Size = %08X",
 			fw_info.serdes_lfs_partition_offset,
 			fw_info.serdes_lfs_partition_size);
